### ISSUE TITLE
Switcher becomes permanently stuck and focused until gnome session ends

### ISSUE THREAD

**Issue by smcclennon** [daniellandau/switcher#152]
## Environment
**OS:** Fedora Workstation 35
**Kernel:** 5.16.12-200.fc35.x86_64
**gnome-session:** 41.3-1.fc35
**Window manager:** Wayland
**Switcher version:** 34

**Additional info:**
- VPN with kill switch enabled, so on login there is no internet connection. Tests were also performed with kill switch disabled, results were identical.
- All of my tests were performed with only the Switcher gnome extension enabled.

## Bug description
On a new session, after opening any application, pressing the Switcher keyboard shortcut displays the Switcher search bar with no drop-down results. The search bar can be typed in to, but no results appear. The search bar cannot be exited with `Esc`, `Alt + F4`, or clicking elsewhere.

Switcher remains focused regardless of where on the screen you click. The panel (top bar) items cannot be clicked.

I can enter the activities overview (list of desktop workspaces) by pressing the `super` key, but cannot click anything except the Switcher bar which remains overlayed over everything in the same position.

I can still focus on things using the keyboard, for example pressing `super + 9` to open my 9th pinned application, and typing directly in to the activities overview and pressing `Enter` to launch apps. Once apps are launched I can also `Tab` and `Enter` to interact with things such as item check boxes, and can use a terminal.

The only way I know to recover from this state is pressing `Alt + f2` and typing in `gnome-session-quit --logout --no-prompt` and pressing the enter key.

## Steps to reproduce
1. Log in at the gnome login screen.
2. Press the Switcher keyboard shortcut, observe Switcher pops up and functions correctly, closing when defocused
3. Open an app such as `gnome-calculator`.
4. Press the Switcher keyboard shortcut again, observe the Switcher search bar appear with no dropdown results, and no way to close it (even if disabling the extension).

## Switcher Settings
I changed my Switcher keyboard shortcut from `Ctrl + W` to `Ctrl + E` during my troubleshooting process.
![image](https://user-images.githubusercontent.com/24913281/157640551-4de0d9f0-5530-4807-9828-e7e6ea8417c6.png)|

## Screenshots
- Just logged in to Gnome and have not opened any applications yet. Pressed `Ctrl + E`. Switcher performing correctly.
![image](https://user-images.githubusercontent.com/24913281/157645370-d3e2b53c-2f98-4eaf-aec7-41496782f1c1.png)

- Closed Switcher. Launched "gnome-calculator" and then pressed `Ctrl + E` again. Switcher is now in the stuck state.
![image](https://user-images.githubusercontent.com/24913281/157645696-d00cb3b5-25b6-40f0-92a7-98c6c2177293.png)

- Pressed `super` to enter the activity overview. Switcher is still overlaying, and I cannot click anything.
![image](https://user-images.githubusercontent.com/24913281/157645754-5a5d3f74-bb11-4d60-9052-e43100929522.png)

- Pressed `super` to close the activity overview. Clicked on the Switcher search bar and typed inside.
![image](https://user-images.githubusercontent.com/24913281/157645955-fca70af5-1b6d-450c-b610-6f721d32a089.png)

- Pressed `Alt + Tab` to focus `gnome-calculator`. Typed inside it.
![image](https://user-images.githubusercontent.com/24913281/157646153-cb6b09fc-de83-4f43-a20c-5c300e7136e9.png)

- Pressed `super` twice to enter the applications menu. Clicked on the Switcher search bar and typed inside.
![image](https://user-images.githubusercontent.com/24913281/157646341-4f504491-fdbe-4acd-8615-8d597ea75230.png)

- Pressed `super` to enter the activity overview. Typed in "extensions" and pressed `enter` to open the Gnome Extensions application. Pressed the `tab` key until the checkbox for disabling Switcher was selected. Pressed `enter` to disable Switcher. Observe the Switcher search bar persist but visually change.
![image](https://user-images.githubusercontent.com/24913281/157647413-d54d1b40-e38b-4830-b3af-7b666c4ef5ae.png)

- Pressed `enter` on the checkbox again to re-enable Switcher.
![image](https://user-images.githubusercontent.com/24913281/157647613-8df4faf4-fa20-4e30-8388-2f170f865db1.png)

- Pressed `Alt +F2` to open the "Run a command" popup. Typed in `gnome-session-quit --logout --no-prompt` to close my Gnome session and return to the login screen, to get rid of the Switcher search bar.
[ No screenshot ]

**Comment by smcclennon** [comment]
## Journal error logs

Error logs are produced when Switcher enters this "stuck" state, which can be found using `journalctl --grep "switcher@landau.fi"`:

```
Mar 12 18:18:17 hostname gnome-shell[53448]: JS ERROR: ReferenceError: can't access lexical declaration 'debouncedActivateUnique' before initialization
                                           rerunFiltersAndUpdate@/home/username/.local/share/gnome-shell/extensions/switcher@landau.fi/extension.js:206:7
                                           _showUI@/home/username/.local/share/gnome-shell/extensions/switcher@landau.fi/extension.js:217:24
                                           enable/<@/home/username/.local/share/gnome-shell/extensions/switcher@landau.fi/extension.js:493:13
Mar 12 18:18:18 hostname gnome-shell[53448]: JS ERROR: ReferenceError: can't access lexical declaration 'i' before initialization
                                           showSingleBox@/home/username/.local/share/gnome-shell/extensions/switcher@landau.fi/extension.js:457:9
                                           setTimeout/<@/home/username/.local/share/gnome-shell/extensions/switcher@landau.fi/util.js:14:5
```

## Further troubleshooting
After logging in to a new gnome session, I opened a terminal and left it alone for 2 hours. Screen saver was disabled, and the VPN killswitch prevented any internet connections. After 2 hours has passed, I pressed `Ctrl + E` and switcher appeared in the frozen state. This produced the error above in the journal logs again.

**Comment by daniellandau** [comment]
Good catch and thank you for the detailed report! I'll try to get an actual fix done as soon as possible, but as a workaround you can disable "Immediate Activation" to prevent the crash in the current versions. It only happens when you have exactly one window open and "Immediate Activation" is enabled.

**Comment by ahouseago** [comment]
I'm also seeing the same thing on Fedora Workstation 36 using Gnome 42.4 but "Immediate Activation" is false, so that workaround doesn't seem to work any more.

**Comment by brettk-git** [comment]
I am encountering the exact same bug, and see the same error message in the log.
If you cannot fix this soon, please at least provide a way to close the Switcher window and get unstuck without having to quit the whole session.

**Comment by plantimals** [comment]
I have the same issue. Running gnome on wayland from nixos. I am happy to provide more details if it would help.

**Comment by yyagin** [comment]
I’m experiencing the same issue. I’m running Arch with GNOME 48 on X11.
There are two user accounts on this machine, and I use the extension with both. Interestingly, one account has never encountered the issue, while the other consistently does—particularly right after logging in.

To recover the session, I open the “Run a Command” dialog using `ALT+F2` and enter `r` to restart the GNOME session. Although this restarts the session, the problem persists. However, disabling and re-enabling the switcher extension via the Extension Manager resolves it temporarily.

If there’s anything I can do to help fix this, I’d be happy to contribute.
